nixos/modules/services/monitoring/glpi-agent.nix

```diff
@@ -0,0 +1,118 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.glpiAgent;
+
+  configFile = pkgs.writeText "agent.cfg" ''
+    # Server configuration
+    server = ${lib.concatStringsSep ", " cfg.servers}
+
+    # Configuration
+    ${cfg.extraConfig}
+  '';
+
+in
+{
+  options = {
+    services.glpiAgent = {
+      enable = lib.mkEnableOption "GLPI Agent";
+
+      package = lib.mkPackageOption pkgs "glpi-agent" {
+      };
+
+      servers = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        description = ''
+          The URLs of the GLPI servers to connect to.
+        '';
+        example = [ "https://glpi.example.com/inventory" ];
+      };
+
+      extraConfig = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        description = ''
+          Configuration options that will be written to the configuration file.
+        '';
+        example = ''
+          delaytime = 3600
+          timeout = 180
+          tag = mytag
+          logger = stderr
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users.glpi-agent = {
+      description = "GLPI Agent user";
+      isSystemUser = true;
+      group = "glpi-agent";
+      home = "/var/lib/glpi-agent";
+    };
+
+    users.groups.glpi-agent = { };
+
+    systemd.tmpfiles.settings = {
+      "01-glpi-agent-dirs" = {
+        "/var/lib/glpi-agent" = {
+          d = {
+            mode = "0755";
+            user = "glpi-agent";
+            group = "glpi-agent";
+          };
+        };
+
+        "/var/lib/glpi-agent/var" = {
+          d = {
+            mode = "0755";
+            user = "glpi-agent";
+            group = "glpi-agent";
+          };
+        };
+
+        "/etc/glpi-agent" = {
+          d = {
+            mode = "0755";
+            user = "glpi-agent";
+            group = "glpi-agent";
+          };
+        };
+      };
+
+      "02-glpi-agent-conf" = {
+        "/etc/glpi-agent/agent.cfg" = {
+          C = {
+            mode = "0644";
+            user = "glpi-agent";
+            group = "glpi-agent";
+            argument = toString configFile;
+          };
+        };
+      };
+    };
+
+    systemd.services.glpi-agent = {
+      description = "GLPI Agent";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      serviceConfig = {
+        ExecStart = "${lib.getExe cfg.package} --conf-file /etc/glpi-agent/agent.cfg --vardir /var/lib/glpi-agent/var --daemon --no-fork";
```

Comment:
Actually, why is this not injected directly?

Reply:
Hi,

Thank you for your feedback.
1. Same here Why you want to inject directly ?
2. What would be your suggestion on this ?

Best Regards